### PR TITLE
[aws/users,aws/account-settings] new modules

### PR DESCRIPTION
**Pull request by osterman**
## what
* Add a `users` module
* Add a `account-settings module

## why
* Make it easy to use GitOps to provision users

### CODE REVIEW

File: aws/users/variables.tf [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  description = "Stage (e.g. `prod`, `dev`, `staging`)"
+}
+
+variable "name" {
+  type        = "string"
+  description = "Application or solution name (e.g. `app`)"
+  default     = "terraform"
+}
+
+variable "smtp_username" {
```
Comment (aknysh) [inline aws/users/variables.tf]: looks like all vars below are not in use

File: aws/users/main.tf [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+data "aws_iam_account_alias" "default" {}
+
+output "account_alias" {
+  value = "${data.aws_iam_account_alias.default.account_alias}"
+}
+
+locals {
+  signin_url = "${data.terraform_remote_state.account_settings.signin_url}"
```
Comment (aknysh) [inline aws/users/main.tf]: `terraform fmt`

Comment (aknysh) [reply aws/users/main.tf]: locals not in use?

File: aws/users/variables.tf (hunk 1 above)
Comment (osterman) [reply aws/users/variables.tf]: See https://github.com/cloudposse/root.cloudposse.co/pull/24

File: aws/users/main.tf [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  admin_group = ["${data.terraform_remote_state.root_iam.admin_group}", "${data.terraform_remote_state.root_iam.readonly_group}"]
```
Comment (aknysh) [inline aws/users/main.tf]: should it be just one admin group in the list?

File: aws/users/main.tf (hunk 2 above)
Comment (aknysh) [reply aws/users/main.tf]: are we missing the actual `user` files?

File: aws/users/main.tf (hunk 3 above)
Comment (osterman) [reply aws/users/main.tf]: i struggled with naming. my thinking is that admin has admin & readonly (why shouldn't they). readonly people only have readonly. i'm overloading `admin` which i don't like. but i don't know if it's better to add more terms. feels like it's implied that an admin would get readonly. thoughts?

File: aws/users/main.tf (hunk 2 above)
Comment (osterman) [reply aws/users/main.tf]: See https://github.com/cloudposse/root.cloudposse.co/pull/24

File: aws/users/main.tf [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+data "terraform_remote_state" "root_iam" {
+  backend = "s3"
+
+  config {
+    bucket = "${var.namespace}-${var.stage}-terraform-state"
+    key    = "root-iam/terraform.tfstate"
+  }
+}
+
+data "aws_iam_account_alias" "default" {}
```
Comment (aknysh) [inline aws/users/main.tf]: why use data provider and not take the alias from `"terraform_remote_state" "account_settings"`?

File: aws/users/variables.tf (hunk 1 above)
Comment (aknysh) [reply aws/users/variables.tf]: ok I see

File: aws/users/main.tf (hunk 3 above)
Comment (aknysh) [reply aws/users/main.tf]: why not use the `admin` and `readonly` group names from `data.terraform_remote_state.root_iam`, and then in the account modules add the user to one of the group or to both?

```hcl
groups  = ["${local.admin_group}, ${local.readonly_group}"]
```

Comment (osterman) [reply aws/users/main.tf]: My bad. fixed.

**Pull request merged**